### PR TITLE
Clean up and fix Dev portal's "CMS toolbar"

### PR DESCRIPTION
**Pull request by josemigallas**
The name is rather misleading, but this refers to the right hand bar in the developer portal.

* Fixes the Color Theme picker.
* ~Updates jQuery to 3.5~ Removes usage of jQuery
* Fixes minor bugs and over-complicated code

Before:
Screenshot 2024-05-30 at 17 18 20

After:
Screenshot 2024-05-30 at 17 31 50

Will break QE tests. Fix: https://github.com/3scale-qe/3scale-tests/pull/841

### CODE REVIEW

**Comment by mayorova** [comment]
> It seems to work better in a Firefox incognito window, so maybe it's something related to my session or whatever. But anyway I don't see the theme picker:

@jlledom There is a funny check for the theme picker appearance - it is only shown if the provider was created in the last 2 weeks :grimacing:  https://github.com/3scale/porta/blob/ad18ecd0a96babb4d16bd75c1f5f24ae01d448fa/app/lib/logic/cms.rb#L59

This is how much they (we) give to set up the developer portal styling :upside_down_face:

File: lib/developer_portal/app/views/shared/cms/_toolbar.html.slim [hunk 1]
```diff
@@ -1,120 +1,72 @@
...
+                  li class="pf-c-nav__item"
+                    = link_to 'Draft', url_for(request.query_parameters.merge(cms: :draft)),
+                                       class: "pf-c-nav__link #{'pf-m-current' if draft}"
+                  li class="pf-c-nav__item"
+                    = link_to 'Published', url_for(request.query_parameters.merge(cms: :published)),
+                                           class: "pf-c-nav__link #{'pf-m-current' unless draft}"
+                - else
+                  li class="pf-c-nav__item"
+                    a href="#" class="pf-c-nav__link pf-m-current" aria-current="page"
+                      'Static Built-in Page
```
Comment (mayorova) [inline lib/developer_portal/app/views/shared/cms/_toolbar.html.slim:31]: @josemigallas How to reach this branch? :thinking:

Comment (josemigallas) [reply lib/developer_portal/app/views/shared/cms/_toolbar.html.slim:31]: I think this is some kind of fail-safe they added long time ago (~9 year). I don't think it's possible to reach that branch nowadays, unless we override the condition manually.

Here it is when they added the first condition (April 2013): https://github.com/3scale/system/commit/f24dde7d55e948a9b9fa7b6905469562ad305f38#diff-168a5558ac27f8ed51aadf403fea552095bcfe42ef3c1d73d91e3cf84c3d3455

And here the else clause 5 months after: https://github.com/3scale/system/commit/bc737552febc8dfba05904f952b7d1add2f72e7a

**Review by jlledom — APPROVED** [review]
LGTM

**Review by lvillen — APPROVED** [review]

**Pull request merged**